vercel/data_source_dsync_groups.go

```diff
@@ -0,0 +1,132 @@
+package vercel
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-framework/datasource"
+	"github.com/hashicorp/terraform-plugin-framework/datasource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+	"github.com/vercel/terraform-provider-vercel/v3/client"
+)
+
+var (
+	_ datasource.DataSource              = &dsyncGroupsDataSource{}
+	_ datasource.DataSourceWithConfigure = &dsyncGroupsDataSource{}
+)
+
+func newDsyncGroupsDataSource() datasource.DataSource {
+	return &dsyncGroupsDataSource{}
+}
+
+type dsyncGroupsDataSource struct {
+	client *client.Client
+}
+
+func (d *dsyncGroupsDataSource) Metadata(ctx context.Context, req datasource.MetadataRequest, resp *datasource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_dsync_groups"
+}
+
+func (d *dsyncGroupsDataSource) Configure(ctx context.Context, req datasource.ConfigureRequest, resp *datasource.ConfigureResponse) {
+	if req.ProviderData == nil {
+		return
+	}
+
+	client, ok := req.ProviderData.(*client.Client)
+	if !ok {
+		resp.Diagnostics.AddError(
+			"Unexpected DataSource Configure Type",
+			fmt.Sprintf("Expected *client.Client, got: %T. Please report this issue to the provider developers.", req.ProviderData),
+		)
+		return
+	}
+
+	d.client = client
+}
+
+func (d *dsyncGroupsDataSource) Schema(_ context.Context, req datasource.SchemaRequest, resp *datasource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		Description: "Provides information about DSync groups for a team.",
+		Attributes: map[string]schema.Attribute{
+			"team_id": schema.StringAttribute{
+				Description: "The ID of the team to retrieve DSync groups for.",
+				Required:    true,
```

Comment:
This shouldn't be required. We use the team_id configured on the provider itself if not present. e.g.
```
			"team_id": schema.StringAttribute{
				Optional:      true,
				Computed:      true,
				PlanModifiers: []planmodifier.String{stringplanmodifier.RequiresReplaceIfConfigured(), stringplanmodifier.UseStateForUnknown()},
				Description:   "The team ID to add the project to. Required when configuring a team resource if a default team has not been set in the provider.",
			},
```

Reply:
e.g. this should be fine to do: 
```
provider "vercel" {
   team = "team_xxxx" 
}

data "vercel_dsync_groups" "example" {}
```